front/components/assistant/conversation/input_bar/editor/suggestion.ts

```diff
@@ -1,95 +1,31 @@
-import { compareForFuzzySort, subFilter } from "@app/lib/utils";
-import { GLOBAL_AGENTS_SID } from "@app/types";
-
-export type EditorSuggestion = EditorSuggestionUser | EditorSuggestionAgent;
-
-interface BaseEditorSuggestion {
-  id: string;
-  label: string;
-  pictureUrl: string;
-  description: string;
-}
-
-export interface EditorSuggestionUser extends BaseEditorSuggestion {
-  type: "user";
-}
-
-export interface EditorSuggestionAgent extends BaseEditorSuggestion {
-  type: "agent";
-  userFavorite: boolean;
-}
-
-export const isEditorSuggestionAgent = (
-  suggestion: EditorSuggestion
-): suggestion is EditorSuggestionAgent => suggestion.type === "agent";
+/**
+ * Editor suggestion types and utilities.
+ *
+ * This file now uses the centralized mention module for filtering logic.
+ * Type aliases are provided for backward compatibility.
+ */
+
+// TODO(rcs): remove this
+import { filterAgentSuggestions } from "@app/lib/mentions";
+import type {
+  RichAgentMention,
+  RichMention,
+  RichUserMention,
+} from "@app/types";
+import { isRichAgentMention } from "@app/types";
+
+// Type aliases for backward compatibility with existing code.
+export type EditorSuggestionAgent = RichAgentMention;
+export type EditorSuggestionUser = RichUserMention;
+export type EditorSuggestion = RichMention;
+
+export const isEditorSuggestionAgent = isRichAgentMention;
```

Comment:
we use the new types. I'm keeping this so the diff of this PR is not too big. I'll remove them in a later PR